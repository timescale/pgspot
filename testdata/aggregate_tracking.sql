
-- should warn twice
CREATE OR REPLACE AGGREGATE s1.agg3(int) (sfunc=abc);
CREATE OR REPLACE AGGREGATE s1.agg3(int) (sfunc=abc);

CREATE AGGREGATE s1.agg6(int) (sfunc=abc);
-- should not warn since it was previously created in this script
CREATE OR REPLACE AGGREGATE s1.agg6(int) (sfunc=abc);
-- different schema should warn
CREATE OR REPLACE AGGREGATE s2.agg6(int) (sfunc=abc);
-- should warn because no schema
CREATE OR REPLACE AGGREGATE agg6(int) (sfunc=abc);

-- different signature should warn
CREATE OR REPLACE AGGREGATE s1.agg6(int,int) (sfunc=abc);

-- old style
CREATE AGGREGATE s3.agg14 (BASETYPE=int,SFUNC=int4and,STYPE=int);
-- should not warn since it was previously created in this script
CREATE OR REPLACE AGGREGATE s3.agg14 (BASETYPE=int,SFUNC=int4and,STYPE=int);
-- should warn because of different basetype
CREATE OR REPLACE AGGREGATE s3.agg14 (BASETYPE=int8,SFUNC=int4and,STYPE=int);

-- new style with different signature should warn
CREATE OR REPLACE AGGREGATE s3.agg14(int8) (SFUNC=int4and,STYPE=int);

