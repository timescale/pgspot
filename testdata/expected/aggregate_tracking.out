PS007: Unsafe object creation: s1.agg3(integer) at line 2
PS007: Unsafe object creation: s1.agg3(integer) at line 4
PS007: Unsafe object creation: s2.agg6(integer) at line 9
PS017: Unqualified object reference: agg6 at line 11
PS007: Unsafe object creation: agg6(integer) at line 11
PS007: Unsafe object creation: s1.agg6(integer,integer) at line 13
PS007: Unsafe object creation: s3.agg14(int8) at line 21
PS007: Unsafe object creation: s3.agg14(int8) at line 23

Errors: 7 Warnings: 1 Unknown: 0
